File: xposed-pine/src/org/pixel/customparts/hooks/EdgeEffectHookWrapper.java
package org.pixel.customparts.hooks;

import android.content.Context;
import org.pixel.customparts.core.BaseHook;

public class EdgeEffectHookWrapper extends BaseHook {

    private static final String KEY_ENABLED = "overscroll_enabled";

    public String keySuffix = "_xposed";
    public boolean useGlobalSettings = true;

    @Override
    public String getHookId() {
        return "EdgeEffectHook";
    }

    @Override
    public int getPriority() {
        return 10;
    }

    public void setKeySuffix(String keySuffix) {
        this.keySuffix = keySuffix;
    }

    public void setUseGlobalSettings(boolean useGlobalSettings) {
        this.useGlobalSettings = useGlobalSettings;
    }

    @Override
    public boolean isEnabled(Context context) {
        if (context == null) return true;
        return isSettingEnabled(context, KEY_ENABLED, true);
    }

    @Override
    protected void onInit(ClassLoader classLoader) {
        try {
            EdgeEffectHook.configure(useGlobalSettings, keySuffix);
            EdgeEffectHook.initWithClassLoader(classLoader);
            
            log("EdgeEffect hook initialized (suffix=" + keySuffix + ", global=" + useGlobalSettings + ")");
        } catch (Throwable e) {
            logError("Failed to initialize EdgeEffect hook", e);
        }
    }
}

File: xposed-pine/src/org/pixel/customparts/hooks/EdgeEffectHookWrapper.kt
package org.pixel.customparts.hooks

import android.content.Context
import org.pixel.customparts.core.BaseHook












class EdgeEffectHookWrapper : BaseHook() {
    override val hookId = "EdgeEffectHook"
    override val priority = 10  

    




    var keySuffix: String = "_xposed"

    



    var useGlobalSettings: Boolean = true

    companion object {
        private const val KEY_ENABLED = "overscroll_enabled"
    }

    override fun isEnabled(context: Context?): Boolean {
        if (context == null) return true  
        return isSettingEnabled(context, KEY_ENABLED, true)
    }

    override fun onInit(classLoader: ClassLoader) {
        try {
            
            EdgeEffectHook.configure(useGlobalSettings, keySuffix)
            EdgeEffectHook.initWithClassLoader(classLoader)
            log("EdgeEffect hook initialized (suffix=$keySuffix, global=$useGlobalSettings)")
        } catch (e: Throwable) {
            logError("Failed to initialize EdgeEffect hook", e)
        }
    }
}
